sdk/go/internal/querybuilder/querybuilder.go

```diff
@@ -7,16 +7,54 @@ import (
 	"strings"
 
 	"github.com/Khan/genqlient/graphql"
+	"golang.org/x/sync/errgroup"
 )
 
+// GraphQLMarshaller is an internal interface for marshalling an object into GraphQL.
+type GraphQLMarshaller interface {
+	// XXGraphqlType is an internal function. It returns the native GraphQL type name
+	XXGraphQLType() string
+	// XXGraphqlID is an internal function. It returns the underlying type ID
+	XXGraphQLID(ctx context.Context) (string, error)
+}
+
+type argument struct {
+	id    string
+	value any
+}
+
+func (a *argument) resolve(ctx context.Context) error {
```

Comment:
nit: could wrap this with a `sync.Once`, I think it's possible for users creating branching queries that they resolve in parallel (I think this actually happens in our getting started guide w/ running builds in parallel), in which case this could be called concurrently, may as well only run it once.

Just a performance optimization though, not a blocker if more complicated than just `sync.Once`